flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/metadata/MetadataTestUtil.scala

```diff
@@ -272,7 +272,7 @@ object MetadataTestUtil {
 
     val catalogTable = getCatalogTable(resolvedSchema)
 
-    val typeFactory = new FlinkTypeFactory(new FlinkTypeSystem)
+    val typeFactory = new FlinkTypeFactory()
```

Comment:
Why not saving it in a static class variable?

Reply:
because in the next PR this is going to require `ClassLoader` as parameter